doc/chapters/api/index.rst

```diff
@@ -0,0 +1,111 @@
+
+.. _api:
+
+API
+***
+
+This chapter documents the DBus and C API for interacting with the SoftwareContainerAgent. This is
+mostly useful for integrators.
+
+.. TODO::
+    This chapter is still to be written properly. Some bits and pieces have
+    been but they are under development.
+
+.. _dbus-api:
+
+DBus API
+========
+
+:Path: /com/pelagicore/SoftwareContainer
+:Interface: com.pelagicore.SoftwareContainerAgent
+
+Methods
+-------
+
+Ping
+----
+Ping()
+
+CreateContainer
+---------------
+:containerIDPrefix: string
+:config: string
+
+uint32 containerID CreateContainer(string containerIDPrefix, string config)
+
+SetContainerName
+----------------
+:containerID: uint32
+:containerName: string
+
+void SetContainerName(uint32 containerID, string containerName)
+
+LaunchCommand
+-------------
+:containerID: uint32
+:userID: uint32
+:commandLine: string
+:workingDirectory: string
+:outputFile: string
+:env: map<string, string>
```

Comment:
I actually wanted to be as nonspecific as possible as it's documentation of the API, the map syntax is kind of c++'y. I'd prefer to write it agnostic to the underlying language.